numba/tests/test_random.py

```diff
@@ -1374,6 +1380,92 @@ def test_dirichlet_exceptions(self):
                 str(raises.exception),
             )
 
+class TestRandomNoncentralChiSquare(BaseTest):
+
+    def _check_sample(self, size, sample):
+
+        """Check output structure"""
+        if size is not None:
+            self.assertIsInstance(sample, np.ndarray)
+            self.assertEqual(sample.dtype, np.float64)
+            
+            if type(size) is int:
+                self.assertEqual(sample.shape, (size,))
+            else:
+                self.assertEqual(sample.shape, size)
+        else:
+             self.assertIsInstance(sample, float)
+
+        """Check statistical properties"""
+        for val in np.nditer(sample):
+            self.assertGreaterEqual(val, 0)
+
+    def test_noncentral_chisquare_default(self):
+        """
+        Test noncentral_chisquare(df, nonc, size=None)
+        """
+        cfunc = jit(nopython=True)(numpy_noncentral_chisquare_default)
+        inputs = (
+            (0.5, 1), # test branch when df < 1
+            (1, 5),
+            (5, 1),
+            (100000, 1),
+            (1, 10000),
+        )
+        for input in inputs:
+            df, nonc = input
+            res = cfunc(df, nonc)
+            self._check_sample(None, res)
+            nonc = np.nan
+            res = cfunc(df, nonc) # test branch when nonc is nan
+            self.assertTrue(np.isnan(res))
+
+
+    def test_noncentral_chisquare(self):
+        """
+        Test noncentral_chisquare(df, nonc, size)
+        """
+        cfunc = jit(nopython=True)(numpy_noncentral_chisquare)
+        sizes = (None, (10,), (10, 10))
+        inputs = (
+            (0.5, 1),
+            (1, 5),
+            (5, 1),
+            (100000, 1),
+            (1, 10000),
+        )
+
+        for input, size in itertools.product(inputs, sizes):
```

Comment:
I think you can do:

```suggestion
        for (df, nonc), size in itertools.product(inputs, sizes):
```

And delete the line below.

Reply:
Good idea.